numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -2331,6 +2381,39 @@ npyiter_ass_subscript(NewNpyArrayIterObject *self, PyObject *op,
     return -1;
 }
 
+static PyObject *
+npyiter_self(NewNpyArrayIterObject *self)
```

Comment:
Would be better as `npyiter_enter`m since this also sets flags.

Reply:
fixed in 9a45b3d055